### PR TITLE
browserify bundle of botframework-connector

### PR DESCRIPTION
**Pull request by joshgummersall**
Use browserify to produce a browser friendly bundle of `botframework-connector`. This will allow us to better share code across packages.

- cleans up globals
- use cross-fetch which is essentially an isomorphic import-able version of node-fetch
- build a concatenated browser friendly bundle using browserify
- exorcist for browser bundle source map support
- also fixes @module throughout

Note for history: the somewhat odd `postbuild` command is intentional. We want to export `lib/browser.js` via `package.json` but doing so causes browserify to rewrite `lib/index.js` to `lib/browser.js`. The copy command breaks this cycle while ensuring that the proper entrypoint is used to produce the concatenated browser bundle.

See this Github issue for more details: https://github.com/browserify/browserify/issues/1746#issuecomment-443303321

### CODE REVIEW

File: libraries/botframework-connector/src/globals.ts [hunk 1]
```diff
@@ -1,22 +0,0 @@
...
-// set FormData and Fetch as global functions
-function setGlobals() {
-    const env = (global || window) as any;
-    
-    if (!env.hasOwnProperty("FormData")) { 
-        env.FormData = require("form-data");
-    }
-
-    if (!env.hasOwnProperty("fetch")) {
-        env.fetch = require("node-fetch");
```
Comment (joshgummersall) [inline libraries/botframework-connector/src/globals.ts:18]: No longer necessary using `import fetch from "cross-fetch"` which is isomorphic.

File: libraries/botframework-connector/src/globals.ts [hunk 2]
```diff
@@ -1,22 +0,0 @@
...
- * Copyright (c) Microsoft Corporation. All rights reserved.
- * Licensed under the MIT License.
- */
-
-// set FormData and Fetch as global functions
-function setGlobals() {
-    const env = (global || window) as any;
-    
-    if (!env.hasOwnProperty("FormData")) { 
-        env.FormData = require("form-data");
```
Comment (joshgummersall) [inline libraries/botframework-connector/src/globals.ts:14]: As far as I can tell this was not used anywhere within the `botframework-connector` tree.

File: libraries/botframework-connector/.gitignore [hunk 3]
```diff
@@ -1,5 +1,5 @@
-/**/node_modules
-/**/.vscode
-/**/lib/*
-coverage
 .nyc_output
+.vscode
```
Comment (stevengum) [inline libraries/botframework-connector/.gitignore:2]: I think we can move away from having these individual .gitignores, but that can be tracked and performed in a later work item

**Review by stevengum — APPROVED** [review]
LGTM :shipit:

**Comment by stevengum** [comment]
The remaining approvers are not JS-primary devs and required for the `src/auth/` folder in botframework-connector for docstrings or one minor change. I'm going to squash and merge the PR has their code ownership is not for this type of review.

**Pull request merged**